CMakeLists.txt

```diff
@@ -2762,6 +2762,61 @@ if(SDL_VIDEO)
   endif()
 endif()
 
+find_program(GLSLANG NAMES glslang glslangValidator)
+
+if(SDL_VULKAN AND GLSLANG)
+  set(SDL3_Vulkan_headers)
+
+  function(glslang2vkpixelheader dir name)
+    set(input "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/VULKAN_PixelShader_${name}.hlsl")
+    set(intermediate "${CMAKE_CURRENT_BINARY_DIR}/${dir}/VULKAN_PixelShader_${name}.h")
+    set(output "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/VULKAN_PixelShader_${name}.h")
+    add_custom_command(OUTPUT "${intermediate}"
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/${dir}"
+      COMMAND "${GLSLANG}"
+        -D --sep main -e main -S frag --target-env vulkan1.0
+        --auto-sampled-textures --vn "VULKAN_PixelShader_${name}"
+        -o "${intermediate}"
+        "VULKAN_PixelShader_${name}.hlsl"
+      COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${intermediate}" "${output}"
+      DEPENDS "${input}" "${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt"
+    )
+    set(SDL3_Vulkan_headers "${SDL3_Vulkan_headers};${intermediate}" PARENT_SCOPE)
+  endfunction()
+
+  function(glslang2vkvertexheader dir)
+    set(input "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/VULKAN_VertexShader.hlsl")
+    set(intermediate "${CMAKE_CURRENT_BINARY_DIR}/${dir}/VULKAN_VertexShader.h")
+    set(output "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/VULKAN_VertexShader.h")
```

Comment:
Are the generated headers stable between glslang versions?
If they differ, perhaps we should hide the regeneration behind a `SDL_DEVEL` CMake option?

Reply:
> Are the generated headers stable between glslang versions?

Experimentally: not identical, but if they aren't backwards-compatible then I would have assumed that ought to be considered to be a bug in glslang?

(Analogous: C compiler output isn't stable between compilers or versions, but we don't commit generated object code to git and hide regeneration of it behind a special developer option.)

Reply:
glslang should always emit valid spirv, that's not my concern here.
I'm worrying about people accidentally committing their regenerated headers when doing `git add -A`.
This will/might cause back and forth in pull requests.

Reply:
I see your point. Perhaps it could be regenerated if either: a special option is set, or the generated "source" file already doesn't exist? Would that be viable?

Reply:
That would be nice!
I'm thinking about a `SDL_DEVEL` option, but I'm bad at naming things so that's up to you :)